src/Aspire.Dashboard/Components/Pages/IComponentWithTelemetry.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Telemetry;
+using Microsoft.AspNetCore.Components;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+/// <summary>
+/// Interface for components that would like to opt-in to telemetry events, capturing
+/// 1) initialization
+/// 2) component properties on each render
+/// </summary>
+public interface IComponentWithTelemetry
+{
+    public Guid? InitializeCorrelation { get; set; }
+    public NavigationManager NavigationManager { get; }
+    public DashboardTelemetryService TelemetryService { get; }
+
+    public string ComponentId { get; }
+    public Dictionary<string, AspireTelemetryProperty> GetTelemetryProperties() => [];
+}
+
+public static class IComponentWithTelemetryExtensions
+{
+    public static async Task InitializeComponentTelemetryAsync(this IComponentWithTelemetry component)
+    {
+        await component.TelemetryService.InitializeAsync();
+
+        if (component.TelemetryService.IsTelemetryEnabled)
+        {
+            component.InitializeCorrelation = component.TelemetryService.PostUserTask(
+                TelemetryEventKeys.InitializeComponent,
+                TelemetryResult.Success,
+                properties: new Dictionary<string, AspireTelemetryProperty> {
+                    // Component properties
+                    { TelemetryPropertyKeys.DashboardComponentId, new AspireTelemetryProperty(component.ComponentId) }
+                });
```

Comment:
The captures the "page load" event right? Should there also be something recording in dispose to indicate that someone is leaving the page?

Maybe this situation is where it makes sense to have starting the page event and ending the page event be correlated. That way you can see how long someone was on a page.

Reply:
That's a good idea. Added